servers/rendering/renderer_rd/storage_rd/material_storage.cpp

```diff
@@ -807,8 +807,10 @@ void MaterialStorage::MaterialData::update_textures(const HashMap<StringName, Va
 				if (V->value.is_array()) {
 					Array array = (Array)V->value;
 					if (uniform_array_size > 0) {
-						for (int j = 0; j < array.size(); j++) {
-							textures.push_back(array[j]);
+						for (int j = 0; j < uniform_array_size; j++) {
```

Comment:
```suggestion
						for (int j = 0; j < MIN(uniform_array_size, array.size()); j++) {
```
Instead, and remove the check